ddtrace/opentracer/tracer.py

```diff
@@ -53,7 +55,9 @@ def __init__(self, service_name=None, config=None, scope_manager=None):
         if not self._service_name:
             raise ConfigException('a service_name is required')
 
-        self._scope_manager = ScopeManager()
+        # default to using a threadlocal scope manager
+        # TODO: should this be some kind of configuration option?
+        self._scope_manager = ThreadLocalScopeManager()
```

Comment:
`ThreadLocalScopeManager` is the basic thread-local context way of maintaining the active span. There are (or soon will be) managers for tornado, asyncio, etc [0]. The question I think I meant to propose with the TODO is how we should allow configuration/specification of which scope manager to be used by the tracer.

I'm not sure if it would be better to have the user of the library specify a config option like:
```
{  ..., 'scope_manager': 'tornado' }
```
and then we translate this to using `opentracing.ext.TornadoScopeManager`
or allow them to pass the class itself
```
{  ..., 'scope_manager': opentracing.TornadoScopeManager }
```

We lose the ability to have a plain-text configuration if we use the latter, but lose flexibility for if a new opentracing scope manager is added by using the former.

[0] https://github.com/opentracing/opentracing-python/pull/83

Reply:
Yeah, that's a fun can of worms.

Makes sense now given that context. Definitely don't need to solve in this PR.

Reply:
If you would like to have the configuration plaintext, you can always:
```
{  ..., 'scope_manager': 'opentracing.TornadoScopeManager' }
```

This is a common approach in Django, where you import the module to avoid people importing it in their code, mostly to avoid the side effect of the import at configuration time. In our case it doesn't change so much, so it's more a matter of style. I would say to avoid `tornado` because it means keeping an internal mapping (and I prefer to be explicit). Probably you can just pass the string of the class that we import and initialize in the Tracer constructor.

Thoughts?

Reply:
On second thought, it may not make sense to configure the scope manager through the `config` at all and just leave it to the `scope_manager` argument that can be passed through the tracer initializer.